mitodl/odl-video-service
# Improved upload UI

#### What are the relevant tickets?
Closes #226

#### What's this PR do?
Improves the upload UI to be more seamlessly integrated into the rest of the app

#### How should this be manually tested?
Upload a file via the Dropbox button, make sure the the page reloads after your file has been chosen and submitted

#### Where should the reviewer start?
`CollectionDetailPage.js` and `VideoCard.js`

#### Any background context you want to provide?
- I added [react-dropbox-chooser](https://github.com/sdoomz/react-dropbox-chooser), a very small NPM package, to this project. Unfortunately it uses `React.PropTypes`, which is deprecated and triggers a warning. If this becomes a deal-breaker, we can very easily fork the package and fix the ~8 offending lines.
- It seems like we might want to avoid putting the dropbox key value into the front-end `SETTINGS` variable in some cases, but I'm not exactly sure what those cases are (Users that don't have admin permission for the specific collection being viewed? Users that have no admin permissions for any collections? etc.). For now it gets set for all users.
- I'll wait for word from @roberthouse54 or @pdpinch about the preferred text for error/loading messages.
- I'm still unable to successfully upload locally, hence the screenshot.

#### Screenshots (if appropriate)
![ss 2017-09-15 at 11 30 52](https://user-images.githubusercontent.com/14932219/30490893-68110194-9a09-11e7-8877-4adbe7149f35.png)
